ApplicationLibCode/Application/Tools/RiaCurveMerger.inl

```diff
@@ -79,8 +79,15 @@ void RiaCurveMerger<XValueType>::addCurveData( const std::vector<XValueType>& xV
         {
             if ( m_isXValuesSharedBetweenCurves )
             {
-                const auto& firstXValues       = m_originalValues.front().first;
-                m_isXValuesSharedBetweenCurves = std::equal( firstXValues.begin(), firstXValues.end(), xValues.begin() );
+                const auto& firstXValues = m_originalValues.front().first;
+                if ( firstXValues.size() == xValues.size() )
+                {
+                    m_isXValuesSharedBetweenCurves = std::equal( firstXValues.begin(), firstXValues.end(), xValues.begin(), xValues.end() );
+                }
+                else
```

Comment:
Just FYI, if you use the four-argument version of `std::equals()`&ndash;i.e., with a two begin/end iterator pairs&ndash;then you get the size check built into the `equals()` predicate.  The common way to use this is to either explicitly check the size and then call the three-argument overload of `equals()` or to call the four-argument version of `equals()` and not check the size before the call.

Reply:
Thanks for you input. I was able to further simplify the logic.